src/services/TileService.ts

```diff
@@ -431,7 +431,7 @@ export class TileService {
         const compressedArray = tile.imageData;
         const workerIndex = this.compressionRequestCounter % this.workers.length;
         const nanEncodings32 = new Int32Array(tile.nanEncodings.slice(0).buffer);
-        let compressedView = new Uint8Array(tile.width * tile.height * 4);
+        let compressedView = new Uint8Array(compressedArray.byteLength > tile.width * tile.height * 4 ? compressedArray.byteLength : tile.width * tile.height * 4);
```

Comment:
wouldn't `Math.max(compressedArray.byteLength, tile.width * tile.height * 4)` be neater?

Reply:
actually, can we not always just use the byte length? For larger tiles, surely the compressed length (`compressedArray.btyteLength`) will be much smaller than the uncompressed length (`tile.width * tile.height * 4`)?

Reply:
yes, using byte length is better.